draft-ietf-quic-qpack.md

```diff
@@ -188,30 +188,26 @@ input header list.
 QPACK is designed to contain the more complex state tracking to the encoder,
 while the decoder is relatively simple.
 
-### Reference Tracking
-
-An encoder MUST ensure that a header block which references a dynamic table
-entry is not processed by the decoder after the referenced entry has been
-evicted.  Hence the encoder needs to retain information about each compressed
-header block that references the dynamic table until that header block is
-acknowledged by the decoder; see {{header-acknowledgement}}.
-
 ### Limits on Dynamic Table Insertions {#blocked-insertion}
 
 Inserting entries into the dynamic table might not be possible if the table
 contains entries which cannot be evicted.
 
 A dynamic table entry cannot be evicted immediately after insertion, even if it
 has never been referenced. Once the insertion of a dynamic table entry has been
-acknowledged and there are no outstanding unacknowledged references to the
-entry, the entry becomes evictable.
+acknowledged and there are no outstanding references to the entry in
+unacknowledged header blocks, the entry becomes evictable.  Note that
+unacknowledged references on the encoder stream do not preclude the eviction of
```

Comment:
I don't know that we need to say this, since the previous sentence explicitly says 'references to the entry in unacknowledged header blocks' and header blocks do not appear on the encoder stream.

Or maybe the sentence should just indicate that headers on the encoder stream that reference other headers do not count as a reference.

Reply:
I have a slight preference of adding an explicit note that encoder stream references do not count, to make sure no one unnecessarily implements tracking for them (like I foolishly did).  But I'm flexible as far as wording is concerned, and do not insist on adding this note if you don't believe it adds any value.  Let me know what you think.